web/angular-wallet/src/app/layout/components/burst-recipient-input/burst-recipient-input.component.ts

```diff
@@ -134,4 +133,44 @@ export class BurstRecipientInputComponent implements OnInit {
   getValidationClass(): string {
     return 'badge ' + this.recipient.status.toString().toLocaleLowerCase();
   }
+
+  parseQR(): void {
+    const file = this.file.nativeElement.files[0];
+
+    if (file) {
+        const img = new Image();
+        img.src = window.URL.createObjectURL(file);
+        img.onload = () => {
+            const width = img.naturalWidth,
+                  height = img.naturalHeight;
+
+            const reader = new FileReader();
+            reader.onload = (e: ProgressEvent): void => {
+
+              const canvas = document.createElement('canvas');
+              canvas.width = width;
+              canvas.height = height;
+              const ctx = canvas.getContext('2d');
+              ctx.drawImage(img, 0, 0);
+              const { data }  = ctx.getImageData(0, 0, width, height);
+              const qr = jsQR(data, width, height);
```

Comment:
what happens, if that failed - any fallback?

Reply:
No but I was thinking I should show an error message. It returns null if it fails to find a QR code.